### DISCUSSION TITLE
Problem with RSA_PRIVATE_KEY

### DISCUSSION THREAD

**Discussion by xaxalex** [saleor/saleor#9303] [Deployment] [answered]
I want to load RSA_PRIVATE_KEY but I get an error:
 
File "/app/saleor/core/jwt_manager.py", line 155, in validate_configuration
     raise ImproperlyConfigured(f"Unable to load provided PEM private key. {e}")

my key in backend.env looks like this:
RSA_PRIVATE_KEY=./saleor.pem

The file with the key is in the folder with the backend.env file.

Please tell me how to set this variable correctly

**Comment by carlosa54** [comment, marked as answer]
Hey @xaxalex you need to set the actual value of the cert instead of the file path.

E.g.
```RSA_PRIVATE_KEY="-----BEGIN RSA PRIVATE KEY-----{....}-----END RSA PRIVATE KEY-----"```

**Comment by jbergstroem** [comment]
I think a nice improvement would be for saleor to check if the input is a path to a file (or: not a private key), and if so; try to read it!

**Comment by xaxalex** [comment]
Thank you very much!

I tried this option, but it gave an error.
Now I understand why. I used far2l to edit the backend.env file.
Now I did it through VS Code and it all worked.
